### ISSUE TITLE
Reduce number of calls to ZAMG servers

### ISSUE THREAD

**Issue by Langdi** [killer0071234/python-zamg#104]
## Version of zamg

0.2.1

## Describe the bug

Unfortunately, I cannot provide a detailed description, but please check your logic for calls to ZAMG servers. We receive calls to both the metadata and data endpoints far more often than every 10 min. With an increasing number of users of this plugin, this is putting a heavy load on our servers. 

## Debug log

Not applicable.

**Comment by killer0071234** [comment]
Thank's for your information.
Currently home assistant is using a old version of this integration which has a wrong parsing of the timestamp (api is responding without seconds) and due that it is trying to restart the integration over and over.

**Comment by Langdi** [comment]
I see. Thanks for your cooperation. It is probably also helpful to have more meaningful information about the plugin/user-agent. Please include an URL (e.g. this github repo) and the version number of this integration. (Other best practices regarding user agents: https://developers.whatismybrowser.com/learn/browser-detection/user-agents/user-agent-best-practices) This can help with analyzing such situations.

**Comment by steinerl** [comment]
@Langdi do you work for ZAMG? In what role?
Is there a particular reason why the timestamp format was changed, especially without notifying users first or without a changelog of any kind, or did we miss something?
Is there a way of direct communication to avoid such problems in the future?

**Comment by killer0071234** [comment]
> > do you work for ZAMG? In what role?
> 
> Yes, I'm a developer working on the API project.
> 
> > Is there a particular reason why the timestamp format was changed, especially without notifying users first or without a changelog of any kind, or did we miss something?
> 
> The reason this was changed is, that there is no data in second resolution. Therefore it was deemed not necessary and potentially confusing for users (this does not really apply when using the API directly, but rather when used via our frontend). You did not miss anything, this was not announced in advance.
> 
> > Is there a way of direct communication to avoid such problems in the future?
> 
> As of now there is only [datahub.feedback@zamg.ac.at](mailto:datahub.feedback@zamg.ac.at). We will think about how to have a meaningful 2-way communication (so we can also communicate changes in advance).

@Langdi  Thanks for your informations.
One possible way to communicate changes to the public is to put it at least [here](https://dataset.api.hub.zamg.ac.at/v1/docs/changelog.html) on the api docs.

**Comment by killer0071234** [comment]
> I see. Thanks for your cooperation. It is probably also helpful to have more meaningful information about the plugin/user-agent. Please include an URL (e.g. this github repo) and the version number of this integration. (Other best practices regarding user agents: https://developers.whatismybrowser.com/learn/browser-detection/user-agents/user-agent-best-practices) This can help with analyzing such situations.

Yes that's right.
I will put it on the list of changes ;-)

**Comment by steinerl** [comment]
> Yes, I'm a developer working at the API project.

I see. Thanks for clarifying.

> The reason this was changed is, that there is no data in second resolution. Therefore it was deemed not necessary and potentially confusing for users (this does not really apply when using the API directly, but rather when used via our frontend). You did not miss anything, this was not announced in advance.

I can understand the reasoning behind the change, but I don't quite understand why the API is versioned when breaking changes are introduced into existing versions anyway.

> As of now there is only [datahub.feedback@zamg.ac.at](mailto:datahub.feedback@zamg.ac.at). We will think about how to have a meaningful 2-way communication (so we can also communicate changes in advance).

Like @killer0071234 said, it would be good if at least the changelog was maintained so that one can be sure that the changes are permanent and were not introduced accidentally.

**Comment by dlang-geosphereat** [comment]
Sorry, for the late replies to this.

> I can understand the reasoning behind the change, but I don't quite understand why the API is versioned when breaking changes are introduced into existing versions anyway.

I feel your frustration. We did not treat it as a breaking change, because we served ISO 8601 timestamps before and did not specify any further. Since it **did** break stuff, we should have treated this as a breaking change.

We also updated the changelog in this regard (quite some time ago).

Anyhow, this issue is resolved and can be closed. (I'll do so, using my private account, which opened this issue)
